src/check.h.in

```diff
@@ -747,6 +747,108 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
  */
 #define ck_assert_str_ge(X, Y) _ck_assert_str(X, >=, Y)
 
+/* Memory location comparison macros with improved output compared to ck_assert() */
+/* OP might be any operator that can be used in '0 OP memcmp(X,Y,L)' comparison */
+/* The x and y parameter swap in memcmp() is needed to handle >, >=, <, <= operators */
+/* Output is limited to 64 bytes (or 128 hex digits) */
+#define _ck_assert_mem(X, OP, Y, L) do { \
+  const char* _ck_x = (const char*)(void*)(X); \
+  const char* _ck_y = (const char*)(void*)(Y); \
+  size_t _ck_l = (L); \
+  char _ck_x_str[129]; \
+  char _ck_y_str[129]; \
+  static char _ck_hexdigits[] = "0123456789abcdef"; \
+  size_t _ck_i; \
+  for (_ck_i = 0; _ck_i < ((_ck_l > 64) ? 64 : _ck_l); _ck_i++) { \
+    _ck_x_str[_ck_i * 2  ]   = _ck_hexdigits[(_ck_x[_ck_i] >> 4) & 0xF]; \
+    _ck_y_str[_ck_i * 2  ]   = _ck_hexdigits[(_ck_y[_ck_i] >> 4) & 0xF]; \
+    _ck_x_str[_ck_i * 2 + 1] = _ck_hexdigits[_ck_x[_ck_i] & 0xF]; \
+    _ck_y_str[_ck_i * 2 + 1] = _ck_hexdigits[_ck_y[_ck_i] & 0xF]; \
+  } \
+  _ck_x_str[_ck_i * 2] = 0; \
+  _ck_y_str[_ck_i * 2] = 0; \
+  ck_assert_msg(0 OP memcmp(_ck_y, _ck_x, _ck_l), \
+    "Assertion '"#X#OP#Y"' failed: "#X"==\"%s\", "#Y"==\"%s\"", _ck_x_str, _ck_y_str); \
```

Comment:
I am torn between the message containing "#X#OP#Y" or something else which mentions the size of the strings. Should the message mention memcmp, as the documentation uses? For example, "0#OPmemcmp(#X,#Y,#L)"?